lib/chef/knife/ec_base.rb

```diff
@@ -120,110 +128,128 @@ def org_admin
+        end
 
       def remote_users
         @remote_users ||= rest.get('/users')
-      end
+        end
 
       def remote_user_list
         @remote_user_list ||= remote_users.keys
-      end
+        end
 
       def local_user_list
         @local_user_list ||= Dir.glob("#{dest_dir}/users/*\.json").map { |u| File.basename(u, '.json') }
-      end
+        end
 
       def users_for_purge
         # not itended to be called from ec_base
         raise Chef::Knife::EcBase::UnImplemented
-      end
+        end
 
       def knife_ec_error_handler
         @knife_ec_error_handler ||= Chef::Knife::EcErrorHandler.new(dest_dir, self.class)
-      end
+        end
 
       def user_acl_rest
         @user_acl_rest ||= if config[:skip_version]
                              rest
                            elsif server.supports_user_acls?
                              rest
                            elsif server.direct_account_access?
-                             Chef::ServerAPI.new("http://127.0.0.1:9465", {:api_version => "0"})
-                           end
-
+                             Chef::ServerAPI.new('http://127.0.0.1:9465', api_version: '0')
       end
+        end
 
       def set_skip_user_acl!
         config[:skip_useracl] ||= !(server.supports_user_acls? || server.direct_account_access?)
-      end
+        end
 
       def set_client_config!
-        Chef::Config.custom_http_headers = (Chef::Config.custom_http_headers || {}).merge({'x-ops-request-source' => 'web'})
+        Chef::Config.custom_http_headers = (Chef::Config.custom_http_headers || {}).merge('x-ops-request-source' => 'web')
         Chef::Config.node_name = 'pivotal'
         Chef::Config.client_key = webui_key
-      end
+        end
 
       def set_dest_dir_from_args!
         if name_args.length <= 0
-          ui.error("Must specify backup directory as an argument.")
+          ui.error('Must specify backup directory as an argument.')
           exit 1
         end
         @dest_dir = name_args[0]
-      end
+        end
+
+      def validate_skip_objects
+        valid_objects = %w[cookbooks cookbook_artifacts nodes clients
+                           roles environments data_bags policies policy_groups
+                           acls groups users useracls invitations.json members.json]
+
+        self.skip_objects = config[:skip_objects].split(',').uniq
+        config[:skip_users] = true if skip_objects.include?('users')
+        if skip_objects.include?('useracls') || skip_objects.include?('useracl')
+          config[:skip_useracl] = true
+        end
+        skip_objects.map! { |x| %w[invitations members].include?(x) ? "#{x}.json" : x }
+        invalid_objects = skip_objects - valid_objects
+        # self.skip_objects.push('cookbook_artifacts').uniq if skip_objects.include?('cookbooks')
+        if invalid_objects.count > 0
+          ui.error "Invalid values passed to skip_objects: #{invalid_objects}"
+          raise
+        end
+        end
```

Comment:
Looking beyond the two `end` s at the same level, I like what I see here.
I'm going to pull down your branch next and test it out functionally next.